### PR TITLE
Create section 27 "Webhooks" of the Controller UG

### PR DESCRIPTION
**Pull request by EMcWhinn**
Adding assembly and modules

Create section 27 of the Controller user guide

https://issues.redhat.com/browse/AAP-16683

Affects `titles/controller-user-guide`

### CODE REVIEW

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+= Setting up a GitHub webhook
+
+{ControllerNameStart} has the ability to run jobs based on a triggered webhook event coming in. 
+Job status information (pending, error, success) can be sent back only for pull request events. 
+If you do not need {ControllerName} to post job statuses back to the webhook service, go directly to step 3.
+
+.Procedure
+
+. Generate a personal access token (PAT) for use with {ControllerName}:
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:11]: _ Personal Access Token _ (PAT)

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+.. In the profile settings of your GitHub account, select *Settings*.
+.. From the navigation panel, select menu:<> Developer Settings[].
+.. In the *Developer Settings*, select *Personal access tokens*.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:14]: On the * Developer Settings * page, select...

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+.. When prompted, enter your GitHub account password to continue.
+.. In the *Note* field, enter a brief description about what this PAT is used for.
+.. In the *Select scopes* fields, check the boxes next to *repo:status*, *repo_deployment*, and *public_repo*. 
+The automation webhook only needs repository scope access, with the exception of invites.
+For more information, see link:https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps[Scopes for OAuth apps documentation].
+.. Click btn:[Generate token].
+.. When the token is generated, ensure that you copy the PAT, as you need it in step 2. 
+You cannot access this token again in GitHub.
+. Use the PAT to optionally create a GitHub credential:
+.. Go to your instance, and xref:ref-controller-credential-gitHub-pat[create a new credential for the GitHub PAT] using the generated token.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:25]: [Create a new...

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
++
+image::ug-webhooks-webhook-credential.png[GitLab webhook credential]
++
+.. Click btn:[Save]. Your job template is set up to post back to GitHub.
+. Go to a GitHub repository where you want to configure webhooks and select menu:Settings[].
+. From the navigation panel, select menu:Webhooks[Add webhook].
+. To complete the *Add webhook* page, you must check the *Enable Webhook* option in a job template or workflow job template. 
+For more information, see step 3 in both xref:controller-create-job-template[Creating a job template] and xref:controller-create-workflow-template[Creating a workflow template].
+. Complete the following fields:
+Payload URL:: Copy the contents of the *Webhook URL* from the job template and paste it here.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:40]: Add a blank line and make the following a bulleted list.

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Secret:: Copy the contents of the *Webhook Key* from the job template and paste it here.
+Which events would you like to trigger this webhook?:: Select the types of events you want to trigger a webhook. 
+Any such event will trigger the job or workflow. 
+To have the job status (pending, error, success) sent back to GitHub, you must select *Pull requests* in the *Let me select individual events* section.
++
+image::ug-webhooks-github-repo-choose-events.png[Github repo choose events]
++
+Active:: Leave this checked.
+. Click btn:[Add webhook].
+. When your webhook is configured, it displays in the list of webhooks active for your repository, along with the ability to edit or delete it.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:52]: it is displayed in...

File: downstream/modules/platform/proc-controller-set-up-gitlab-webhook.adoc [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+{ControllerNameStart} has the ability to run jobs based on a triggered webhook event coming in. 
+Job status information (pending, error, success) can be sent back only for pull request events. 
+If you do not need {ControllerName} to post job statuses back to the webhook service, go directly to step 3.
+
+.Procedure
+. Generate a personal access token (PAT) for use with {ControllerName}:
+.. From the navigation panel in GitLab, select your avatar and menu:Edit profile[].
+.. From the navigation panel, select menu:Access tokens[].
+.. Complete the following fields:
+Token name:: Enter a brief description about what this PAT is used for.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-gitlab-webhook.adoc:14]: Blank line and a bulleted list.

File: downstream/modules/platform/proc-controller-set-up-gitlab-webhook.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+Expiration date:: Skip this field unless you want to set an expiration date for your webhook.
+Select scopes:: Select those that are applicable to your integration. 
+For {ControllerName}, *api* is the only selection necessary.
+.. Click btn:[Create personal access token].
+.. When the token is generated, ensure that you copy the PAT, as you need it in step 2. 
+You cannot access this token again in GitLab.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-gitlab-webhook.adoc:20]: Again NOTE or IMPORTANT?

File: downstream/modules/platform/proc-controller-view-payload-output.adoc [hunk 8]
```diff
@@ -0,0 +1,17 @@
+[id="controller-view-payload-output"]
+
+= Viewing the payload output
+
+The entire payload is exposed as an extra variable.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-view-payload-output.adoc:5]: As a result of this procedure?

**Review by ianf77 — COMMENTED** [review]
A few comments

File: downstream/modules/platform/proc-controller-set-up-github-webhook.adoc [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+.. In the *Developer Settings*, select *Personal access tokens*.
+.. From the *Personal access tokens* screen, click btn:[Generate a personal access token].
+.. When prompted, enter your GitHub account password to continue.
+.. In the *Note* field, enter a brief description about what this PAT is used for.
+.. In the *Select scopes* fields, check the boxes next to *repo:status*, *repo_deployment*, and *public_repo*. 
+The automation webhook only needs repository scope access, with the exception of invites.
+For more information, see link:https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps[Scopes for OAuth apps documentation].
+.. Click btn:[Generate token].
+.. When the token is generated, ensure that you copy the PAT, as you need it in step 2. 
+You cannot access this token again in GitHub.
```
[the remark this answers is not in the document]
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-set-up-github-webhook.adoc:23]: I agree

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**